src/index.ts

```diff
@@ -319,6 +323,19 @@ export function ExportExtendedPublicKey({
  * // ["ababab...", // 1 per input]
  *
  */
+interface SignMultisigTransactionArgs {
+  keystore: KEYSTORE_TYPES;
+  network: BitcoinNetwork;
+  inputs?: TxInput[];
+  outputs?: object[];
+  bip32Paths?: string[];
+  psbt: string;
+  keyDetails: { xfp: string; path: string };
+  returnSignatureArray?: boolean;
+  walletConfig: MultisigWalletConfig;
+  policyHmac?: string;
+  progressCallback?: () => void;
+}
```

Comment:
Do you have a preference for `interface` over `type`?

They are SO close to being the same thing now, but the main differences as I understand them are:

- `type` has a more succinct syntax (`&` versus `extends` for example)
- `declaration merging` can only be done with `interface` (although I don't like the practice because its a bit "magical")
- `type` can union interfaces together, but not the other way around...

So generally I reach for `type` unless:
- I specifically need to `extend` a class with an interface or have a desire to allow declaration merging (which is never honestly - but maybe if I was creating a library with exported types I **might** want them to be interfaces to allow consumers to extend them via declaration merging??)

Reply:
Heh, I've actually leaned in the other direction that basically in most cases interfaces are preferable over types simply because I remember a while ago reading that they're mostly the same but the extending in interfaces is useful so just use them unless you really need types. This seems to be the advice in the [TS docs,](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces) although they also say it's basically personal preference unless you really need one feature or the other. The, admittedly simplistic, heuristic I've used is that I like types for more static items, single line declarations, things of that nature, and then interfaces for object and class shapes if that makes sense.

Reply:
Yeah its not very consequential but `&` and `extends` do the same I think.

Anyways, onto important things! lol